redisvl/extensions/sessionmanager/session.py

```diff
@@ -0,0 +1,129 @@
+        LLM responses to allow for enriching future prompts with session
+        context. Session history is stored in prompt:response pairs referred to
+        as exchanges.
+
+        Args:
+            from_file Optional[str]: File to intiaialize the cache index with.
+            vectorizer Vectorizer: The vectorizer to create embeddings with.
+
+        The proposed schema will support a single combined vector embedding constructed
+        from the prompt & response in a single string.
+
+        schema = IndexSchema.from_dict({"index": {"name": name, "prefix": prefix}})
+        schema.add_fields(
+            [
+                {"name": prompt_field, "type": "text"},
+                {"name": response_field, "type": "text"},
+                {
+                    "name": combined_vector_field,
+                    "type": "vector",
+                    "attrs": {
+                        "dims": vectorizer_dims,
+                        "datatype": "float32",
+                        "distance_metric": "cosine",
+                        "algorithm": "flat",
+                 },
+            ]
+        )
+
+
+        """
+        pass
+
+
+    def clear(self):
+        """ Clears the chat session history. """
+        pass
+
+
+    def fetch_context( self, prompt: str, as_text: bool = False, top_k: int = 5) -> Union[str, List[str]]:
+        """ Searches the chat history for information semantically related to
+        the specified prompt.
+
+        This method uses vector similarity search with a text prompt as input.
+        It checks for semantically similar prompt:response pairs and fetches
+        the top k most relevant previous prompt:response pairs to include as
+        context to the next LLM call.
+
+        Args:
+            prompt str: The text prompt to search for in session memory
+            as_text bool: Whether to return the prompt:response pairs as text
+            or as a list.
+            top_k int: The number of previous exchanges to return. Default is 5.
+
+        Returns:
+            Union[str, List[str]: Either a single string, or a list of strings
+            containing the most relevant
+                return fields for each similar cached response.
+
+        Raises:
+            ValueError: If top_k is an invalid integer.
+        """
+        pass
+
+
+    def conversation_history(self, as_text: bool = False) -> Union[str, List[str]]:
+        """ Retreive the full conversation history in sequential order.
+
+        Args:
+            as_text bool: Whether to return the conversation as a single string,
+                          or list of alternating prompts and responses.
+
+        Returns:
+            Union[str, List[str]]: A single string transcription of the session
+                                   or list of strings if as_text is false.
+        """
+        pass
+
+
+    def _order_by(self, exchanges: List[str], recency: bool = True) ->  List[str]:
```

Comment:
Sorting is needed because sometimes we return results based on relevance, and sometimes by recency. It's being done as a final step currently,  but I plan to move it into `query.py` as a parameter.